plugins/metrics/scripts/collector.py

```
@@ -0,0 +1,153 @@
+import threading
+from datetime import datetime, timezone
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+MAX_BODY_SIZE = 1_048_576  # 1 MB
+
+
+class OTLPServer(HTTPServer):
+    allow_reuse_address = True
+
+
+DEFAULT_PID_FILE = "/tmp/claude-otel-collector.pid"
```

Ruff (0.15.17)

[error] Probable insecure usage of temporary file or directory: "/tmp/claude-otel-collector.pid"

(S108)